Tasks/Maven/CodeAnalysis/Common/CodeAnalysisOrchestrator.ts

```diff
@@ -54,17 +57,30 @@ export class CodeAnalysisOrchestrator {
         }
     }
 
-    private processResults(tools: IAnalysisTool[]):AnalysisResult[] {
+    private processResults(tools: IAnalysisTool[]): AnalysisResult[] {
 
         let analysisResults: AnalysisResult[] = [];
 
         for (var tool of tools) {
-            var results:AnalysisResult[] = tool.processResults();
+            var results: AnalysisResult[] = tool.processResults();
             if (results) {
                 analysisResults = analysisResults.concat(results);
             }
         }
 
         return analysisResults;
     }
+
+    private checkBuildContext(): boolean {
+        let requiredVariables: string[] = ['build.sourcesDirectory', 'build.artifactStagingDirectory', 'build.buildNumber'];
+
+        for (var requiredVariable of requiredVariables) {
+            if (!tl.getVariable(requiredVariable)) {
+                tl.debug('[CA] Code analysis is disabled outside of the build enviroment. Could not find a value for ' + requiredVariable);
```

Comment:
just checking `Build.BuildId` is good for build/release platform? Rest of the variables should be there, if not, it's error in build instead of skipping the job?